packages/labs/task/src/test/task_test.ts

```diff
@@ -225,6 +226,36 @@ suite('Task', () => {
     assert.equal(el.taskValue, `a1,b1`);
   });
 
+  test('tasks with args run when args change: deepEqual', async () => {
+    const el = getTestElement({
+      args: () => [[el.a], [el.b]],
+      argsEqual: deepArrayEquals,
+    });
+    await renderElement(el);
+    el.resolveTask();
+    await tasksUpdateComplete();
+    assert.equal(el.task.status, TaskStatus.COMPLETE);
+    assert.equal(el.taskValue, `a,b`);
+
+    // Changing task argument runs task
+    el.a = 'a1';
+    // Check task pending.
+    await tasksUpdateComplete();
```

Comment:
TODO: the name of this function is confusing. It _doesn't_ wait for the task to complete. It just waits a rAF. That's why it can be used to check that a task is pending.